onboard-jmuniz/hooks/useCreateUser.ts

```diff
@@ -0,0 +1,67 @@
+import { FetchResult, gql, useMutation } from "@apollo/client";
+import { useState } from "react";
+
+const LOGIN_USER = gql`
+  mutation CreateUser($data: UserInput!) {
+    createUser(data: $data) {
+      id
+    }
+  }
+`;
+
+export enum UserRole {
+  USER = "user",
+  ADMIN = "admin",
+}
+
+export interface CreateUserProps {
+  email: string;
+  password: string;
+  birthDate: Date;
+  phone: string;
+  name: string;
+  role: UserRole;
+}
+
+export interface UseCreateUserProps {
+  variables: {
+    data: CreateUserProps;
+  };
+}
+export interface CreateUserResponseProps {
+  createUser: {
+    id: number;
+  };
+}
+
+function propsAreValid(props: UseCreateUserProps) {
+  if (
+    props.variables.data.name &&
+    props.variables.data.email &&
+    props.variables.data.password &&
+    props.variables.data.phone &&
+    props.variables.data.birthDate &&
+    Object.values(UserRole).includes(props.variables.data.role)
+  ) {
```

Comment:
A sua validação de formulário já não garante que os valores virão preenchidos?

Reply:
A minha validação checa os valores a cada onChangeText ou onChange e atualiza o campo do state se ele for válido, assim, se a pessoa só apertar o botão sem interagir com os campos, ela faria uma request vazia e por isso eu coloquei essa verificação no fim